File: Telerik/SecondControl/01.ArraySearch.java
package Telerik;

import java.util.*;

public class ArraySearch {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        int[] array = Arrays.stream(scan.nextLine().split("[\\s+,]")).mapToInt(Integer::parseInt).toArray();
        int n = array.length;
        Boolean[] bitmap = new Boolean[n + 1];
        for (int i = 1; i <= n; i++)
        {
            bitmap[i] = false;
        }
        for (int i = 0; i < n; i++)
        {
            bitmap[array[i]] = true;
        }
        List<Integer> missingNumbers = new ArrayList<Integer>();
        for (int i = 1; i <= n; i++)
        {
            if (!bitmap[i])
            {
                missingNumbers.add(i);
            }
        }
        System.out.print(missingNumbers.get(0));
        for (int i = 1; i < missingNumbers.size(); i++) {
            System.out.print("," + missingNumbers.get(i));
        }
    }
}

File: Telerik/SecondControl/01.ArraySearch.cs
using System;
using System.Collections.Generic;
using System.Linq;

class ArraySearch
{
    static void Main()
    {
        int[] array = Console.ReadLine().Split(new[] { ',', ' ' }, StringSplitOptions.RemoveEmptyEntries).Select(int.Parse).ToArray();
        int n = array.Length;
        bool[] bitmap = new bool[n + 1];
        for (int i = 0; i < n; i++)
        {
            bitmap[array[i]] = true;
        }
        List<int> missingNumbers = new List<int>();
        for (int i = 1; i <= n; i++)
        {
            if (!bitmap[i])
            {
                missingNumbers.Add(i);
            }
        }
        Console.WriteLine(string.Join(",", missingNumbers));

    }
}

